lib/srv/discovery/fetchers/db/aws.go

```diff
@@ -17,11 +17,122 @@ limitations under the License.
 package db
 
 import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
 	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	"github.com/gravitational/teleport/lib/srv/discovery/common"
 )
 
+// awsFetcherPlugin defines an interface that provides database type specific
+// functions for use by the common AWS database fetcher.
+type awsFetcherPlugin interface {
+	// GetDatabases fetches databases from AWS API and converts the results to
+	// Teleport types.Databases.
+	GetDatabases(context.Context, *awsFetcherConfig) (types.Databases, error)
+	// ComponentShortName provides the plugin's short component name for
+	// logging purposes.
+	ComponentShortName() string
```

Comment:
What is the component long name ? 
```suggestion
	Name() string
```

Reply:
I'd rather keep it `ComponentShortName` because that's really its whole purpose - to be the short component name in the logger, not the full matcher name